server/src/main/java/org/apache/iotdb/db/metadata/cache/DataNodeSchemaCache.java

```diff
@@ -0,0 +1,223 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.metadata.cache;
+
+import org.apache.iotdb.commons.utils.TestOnly;
+import org.apache.iotdb.db.conf.IoTDBConfig;
+import org.apache.iotdb.db.conf.IoTDBDescriptor;
+import org.apache.iotdb.db.exception.metadata.IllegalPathException;
+import org.apache.iotdb.db.metadata.path.AlignedPath;
+import org.apache.iotdb.db.metadata.path.MeasurementPath;
+import org.apache.iotdb.db.metadata.path.PartialPath;
+import org.apache.iotdb.db.mpp.common.schematree.PathPatternTree;
+import org.apache.iotdb.db.mpp.common.schematree.SchemaTree;
+import org.apache.iotdb.db.mpp.sql.analyze.FakeSchemaFetcherImpl;
+import org.apache.iotdb.db.mpp.sql.analyze.ISchemaFetcher;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.write.schema.IMeasurementSchema;
+import org.apache.iotdb.tsfile.write.schema.MeasurementSchema;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class takes the responsibility of metadata cache management of all DataRegions under
+ * StorageEngine
+ */
+public class DataNodeSchemaCache {
+  private static final Logger logger = LoggerFactory.getLogger(DataNodeSchemaCache.class);
+
+  private static IoTDBConfig config = IoTDBDescriptor.getInstance().getConfig();
+
+  private Cache<PartialPath, SchemaCacheEntity> schemaEntityCache;
+
+  // TODO use fakeSchemaFetcherImpl for test temporarily
+  private static final ISchemaFetcher schemaFetcher = new FakeSchemaFetcherImpl();
+
+  private DataNodeSchemaCache() {
+    schemaEntityCache =
+        Caffeine.newBuilder().maximumSize(config.getDataNodeSchemaCacheSize()).build();
+  }
+
+  public static DataNodeSchemaCache getInstance() {
+    return DataNodeSchemaCache.DataNodeSchemaEntryCacheHolder.INSTANCE;
+  }
+
+  /** singleton pattern. */
+  private static class DataNodeSchemaEntryCacheHolder {
+    private static final DataNodeSchemaCache INSTANCE = new DataNodeSchemaCache();
+  }
+
+  public SchemaCacheEntity getSingleSchemaEntity(
+      PartialPath partialPath, boolean isAutoCreateSchema) {
+    SchemaCacheEntity schemaCacheEntity;
+    if (partialPath instanceof AlignedPath) {
+      return null;
+    }
+    schemaCacheEntity = schemaEntityCache.getIfPresent(partialPath);
+    if (schemaCacheEntity == null) {
+      PartialPath cacheKey;
+      if (partialPath instanceof MeasurementPath) {
```

Comment:
Yes, if we need this partialPath to auto create schema, so the partialPath should contains MeasurmentSchema, and that will be a MeasurementPath.

Reply:
Currently, the ```MeasurementPath``` and ```AlignedPath``` only act as a schema query response for data query, rather than a request. They will be renamed to ```QueryUnit``` or someting like this. There's no need to wrap the params to a measurementPath and then unwrap them.